packages/ckeditor5-dev-release-tools/lib/tasks/preparerepository.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * @license Copyright (c) 2003-2023, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+const upath = require( 'upath' );
+const fs = require( 'fs-extra' );
+
+const pkgJsonTemplatePath = upath.join( __dirname, '..', 'templates', 'release-package.json' );
+
+/**
+ * @param {Object} options
+ * @param {String} [options.cwd] Root of the repository to prepare. `process.cwd()` by default.
+ * @param {String} [options.packagesDir] Path to directory containing packages. `packages` by default.
+ * @param {String} [options.outputDir] Path to the output directory of this script. `release` by default.
+ * @param {Object} options.pkgJsonContent Object containing values to use in the created package json file.
+ * @param {Array<String>} options.rootFilesToCopy Array containing list of files or directories to copy.
+ * @param {Array<String>} [options.packagesToCopy] Optional argument specifying packages to copy from the `packagesDir`.
+ * This can contain nested package paths. If left empty, all non-nested packages from the `packagesDir` will be processed.
```

Comment:
Let's sort these properties from required to optional ones.

```suggestion
 * @param {Object} options
 * @param {String} options.outputDirectory Relative path to the destination directory where packages will be stored.
 * @param {String} [options.cwd] Root of the repository to prepare. `process.cwd()` by default.
 * @param {String} [options.packagesDirectory] Relative path to a location of packages. If specified, all of the found packages will be copied.
 * @param {Array.<String>} [options.packagesToCopy] List of packages that should be processed. If not specified, all packages found in `packagesDirectory` are considered.
* @param {Object} [options.rootPackage] Definition of a root package (found in the `cwd` directory). If not specified, the root package will not be created.
 * @param {Object} options.rootPackage.packageJson Object containing values to use in the created the `package.json` file.
 * @param {Array.<String>} options.rootPackage.files Array containing a list of files or directories to copy.
```

As I am unsure about double nested properties, we can consider creating a typedef describing the root package fields.